### DISCUSSION TITLE
Manually Request Non Cache Version

### DISCUSSION THREAD

**Discussion by jdgamble555** [urql-graphql/urql#1736] [Q&A] [answered]
How Can I use the `stale=true` value in order to manually force a non-cache version.  I only need to do this for **ONE** certain request in my code.

For example in @urql/core:

```typescript
const reset = true;
query = gql`... some query```;

client.query(query, { stale: reset }).toPromise();
```

Is there some variable input to execute something like this?

Thanks,

J

**Comment by kitten** [comment, marked as answer]
I think you are confusing `stale` (which is a property on results) with the `reuestPolicy` https://formidable.com/open-source/urql/docs/basics/document-caching/#request-policies

**Reply by jdgamble555** [reply]
So, how do I get a result without using the cache?  

I only need this for one of my queries and don't need to change the general request policy that uses the cache by default.

J

**Reply by JoviDeCroock** [reply]
Pass `requestPolicy: 'network-only'` in the OperationContext.

**Reply by kitten** [reply]
The `client.query` method takes three arguments:

- The query document or string
- The variables
- and; a partial `context`

As aluded to in the API docs, it's a shortcut to `client.executeQuery`. The `context` here is relevant. It defines _all_ options per operation that make up meta information of the operation itself: https://formidable.com/open-source/urql/docs/api/core/#operationcontext

In other words, options that you pass to the `Client` make their way onto the `context` as defaults, and each individual operation can have slightly altered context options. One of these options is `requestPolicy`. As you can see in the documentation I've linked before, you can alter the `requestPolicy` to skip the cache with `network-only`. This is what this would look like in React: https://formidable.com/open-source/urql/docs/basics/react-preact/#reexecuting-queries

So in the imperative `Client` call you'd instead do:

```js
client.query(
  query,
  { /* variables */ },
  { requestPolicy: 'network-only' },
).toPromise();
```

**Reply by jdgamble555** [reply]
Thank you! That did the trick!
